test/testers.jl

```diff
@@ -530,4 +530,29 @@ end
         test_frule(f_notimplemented, randn(), randn())
         test_rrule(f_notimplemented, randn(), randn())
     end
+
+    @testset "custom rrule_f" begin
+        only2x(x, y) = 2x
+        custom(::typeof(only2x), x, y) = only2x(x, y), Δ -> (NoTangent(), 2Δ, ZeroTangent())
+        wrong1(::typeof(only2x), x, y) = only2x(x, y), Δ -> (ZeroTangent(), 2Δ, ZeroTangent())
+        wrong2(::typeof(only2x), x, y) = only2x(x, y), Δ -> (NoTangent(), 2.1Δ, ZeroTangent())
+        wrong3(::typeof(only2x), x, y) = only2x(x, y), Δ -> (NoTangent(), 2Δ)
+
+        test_rrule(only2x, 2.0, 3.0; rrule_f=custom, check_inferred=false)
+        @test fails(() -> test_rrule(only2x, 2.0, 3.0; rrule_f=wrong1, check_inferred=false))
+        @test fails(() -> test_rrule(only2x, 2.0, 3.0; rrule_f=wrong2, check_inferred=false))
+        @test_broken fails(() -> test_rrule(only2x, 2.0, 3.0; rrule_f=wrong3, check_inferred=false))
```

Comment:
```suggestion
        @test fails(() -> test_rrule(only2x, 2.0, 3.0; rrule_f=wrong3, check_inferred=false))
```
^ This is what it should be, really, but I can't tell why the `fails` doesn't work as expected in this case. (`test_rrule` actually does fail here, as it should, but `fails` does not catch that). Perhaps something to do with the recent `ExprAndMsg` hack @oxinabox?

Reply:
I will investigate

Reply:
This is a bug in ChainRules
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/167

I will fix tomorrow